PMSCore/ReportManager.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System;
+using PMSCore;
+
+namespace PMSCore
+{
+    /// <summary>
+    /// This class handles the creation and management of reports.
+    /// </summary>
+    public sealed class ReportManager
+    {
+        // The single instance of ReportManager
+        private static ReportManager _instance;
+
+        // Logger instance
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Private constructor to prevent external instantiation.
+        /// </summary>
+        private ReportManager()
+        {
+            // Initialize the logger here
+            _logger = LoggerFactory.CreateLogger<ReportManager>();
```

Comment:
CreateLogger requires a name being passed as parameter

Reply:
@DistroRickDev , makes sense, can you please check if I have done it correctly now?